fiona/ogrext.pyx

```diff
@@ -1238,6 +1239,18 @@ cdef class Iterator:
         else:
             OGR_L_SetSpatialFilter(cogr_layer, NULL)
 
+        if where:
+            where_b = where.encode('utf-8')
+            where_c = where_b
+            try:
+                exc_wrap_int(
+                    OGR_L_SetAttributeFilter(cogr_layer, <const char*>where_c))
+            except CPLE_AppDefinedError as e:
+                raise AttributeFilterError(e) from None
```

Comment:
a wee note here, as this is a new Python feature to me [documented here](https://docs.python.org/3/tutorial/errors.html#exception-chaining) and [shown here](https://stackoverflow.com/q/52725278/). It shortens the repetition in the exception trace, without "During handling of the above exception...", but importantly it keeps the original exception message from OGR telling the user what's wrong.

Reply:
😲